### PR TITLE
EVM Precompiles support

### LINKED ISSUE #496: EVM precompiles support
### Problem

The Hedera network supports precompiles as part of EVM and Solidity equivalence.
Occasionally items have shown up that surfaced inequivalence at the language or opcode layer.

### Solution

Add examples for [precompile](https://www.evm.codes/precompiled) operations

- Add `/contracts/evm/Precompiles.sol`
- Cover the tasklist below for each precompile add a function and supporting tests

```[tasklist]
### Panic Error Tasks
- [x] `0x1`: `ecRecover` - it's okay to dupe the logic under the math contract in the repo or as par tof the test class call that contract
- [x] `0x2`: `SHA2-256`
- [x] `0x3`: `RIPEMD-160`
- [x] `0x4`: `identity`
- [x] `0x5`: `modexp`
- [x] `0x6`: `ecAdd`
- [x] `0x7`: `ecMul`
- [x] `0x8`: `ecPairing`
- [x] `0x9`: `Blake2f`
```

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ebadiere**
EVM Precompiles support.

Fixes #496

### CODE REVIEW

File: contracts/evm/Precompiles.sol [hunk 1]
```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (Nana-EC) [inline contracts/evm/Precompiles.sol:1]: Apache license

File: test/evm/Precompiles.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
+/*-
+ *
+ * Hedera JSON RPC Relay - Hardhat Example
```
Comment (Nana-EC) [inline test/evm/Precompiles.js:3]: Wrong header

**Review by Nana-EC — COMMENTED** [review]
Nice work, Some nits

File: contracts/evm/Precompiles.sol (hunk 1 above)
Comment (ebadiere) [reply contracts/evm/Precompiles.sol:1]: Done.

File: test/evm/Precompiles.js (hunk 2 above)
Comment (ebadiere) [reply test/evm/Precompiles.js:3]: Updated.

File: package.json [hunk 3]
```diff
@@ -46,6 +46,8 @@
   },
   "dependencies": {
     "@nomicfoundation/solidity-analyzer": "^0.1.0",
-    "dotenv": "^16.3.1"
+    "bn.js": "^5.2.1",
```
Comment (Nana-EC) [inline package.json:49]: Q: shouldn't this result in an update to the package.lock file also?

File: test/evm/Precompiles.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        const t = ["0x0300000000000000", "0x0000000000000000"];
+
+        const f = true;
+
+        const result = await precompilesContract.callStatic.blake2(rounds, h, m, t, f)
+
+        expect(result[0]).to.equal("0xba80a53f981c4d0d6a2797b69f12f6e94c212f14685ac4b74b12bb6fdbffa2d1")
+        expect(result[1]).to.equal("0x7d87c5392aab792dc252d5de4533cc9518d38aa8dbf1925ab92386edd4009923")
+    });
+
```
Comment (Nana-EC) [inline test/evm/Precompiles.js:207]: nit: I see this extra line before the final closing bracket in many places

**Review by Nana-EC — COMMENTED** [review]
Looking good.
One missing file and a nit

File: test/evm/Precompiles.js (hunk 4 above)
Comment (ebadiere) [reply test/evm/Precompiles.js]: Updated.

File: package.json (hunk 3 above)
Comment (ebadiere) [reply package.json:49]: Added.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**